merqube_client_lib/session.py

```diff
@@ -294,6 +294,9 @@ def get_collection_single(self, url: str, options: Optional[dict[str, Any]] = No
         res = self.get_collection(url, options=options, **kwargs)
         num_results = len(res)
         if num_results != 1:
+            logger.error(f"Expected 1 result but {num_results} found!")
+            if num_results == 0:
+                raise APIError(404, {"message": "No results found for url {url} but 1 was expected!"})
             raise APIError(400, {"message": f"Only one result was expected but {num_results} found!"})
```

Comment:
its a tough one because if the user does get_collection_single, they are asserting there should only be one result. 

i think a 409 might actually be the best